src/nexus_api_client/cli.clj

```diff
@@ -1,12 +1,315 @@
+            [babashka.http-client :as http])
+  (:import [java.io PushbackReader]
+           [java.util.regex Pattern]))
 
-(defn hello-cli
-  [args]
-  (println "Hello" args))
+(defn remove-internal-meta
+  "Removes keywords namespaced with :contajners. They are for internal use."
+  [data-seq]
+  (remove #(= "contajners" (namespace %)) data-seq))
 
+(defn bail-out
+  [^String message]
+  (throw (IllegalArgumentException. message)))
+
+(defn load-api
+  "Loads the API EDN file from resources."
+  []
+  (if-let [config "sonatype-nexus/api.edn"]
+    (-> (io/resource config)
+        (io/reader)
+        (PushbackReader.)
+        (edn/read))
+    (bail-out "Cannot load api, the engine, version combo may not be supported.")))
+
+(defn ops
+  "Returns the supported operations for sonatype nexus (v1) API."
+  [{:keys [v1]}]
+  (->> v1
+       (keys)
+       (remove-internal-meta)))
+
+(defn doc
+  "Returns essential information about the operation."
+  [{:keys [v1]} op]
+  (let [url "http://localhost:8081/service/rest"
+        path (some-> v1 op (:path op))]
+    (some-> v1
+            op
+            (select-keys [:method :path :params :summary])
+            (assoc :doc-url (str url path)))))
+
+(defn create-query
+  [m]
+  (str/join "&" (map #(str (name (key %)) "=" (val %)) m)))
+
+(defn gather-params
+  "Reducer fn categorizing the params as :header, :query or :path.
+  supplied-params: map of params the user has passed when invoking.
+  request-params: accumulator for all the params to be actually sent.
+  each param in the spec is passed as the last arg and categorized by `in` if it is supplied."
+  [supplied-params request-params {:keys [name in]}]
+  (let [param (keyword name)]
+    (if-not (contains? supplied-params param)
+      request-params
+      (update-in request-params [(keyword in)] assoc param (param supplied-params)))))
+
+(defn interpolate-path
+  "Replaces all occurrences of {k1}, {k2} ... with the value map provided.
+  Example:
+  given a/path/{id}/on/{not-this}/root/{id} and {:id hello}
+  results in: a/path/hello/{not-this}/root/hello."
+  [path value-map]
+  (let [[param value] (first value-map)]
+    (if (nil? param)
+      path
+      (recur (str/replace path
+                          (re-pattern (format "\\{([%s].*?)\\}"
+                                              (-> param
+                                                  name
+                                                  Pattern/quote)))
+                          (str value))
+             (dissoc value-map param)))))
+
+(defn api-request
+  ([method url]
+   (api-request method url nil))
+  ([method url opts]
+  ;;  (println method url opts)
+   (try (http/request
+         (merge {:method method :uri (str url)} opts))
+        (catch java.lang.IllegalStateException e 
+          (println "Bad request: \n method: " method "\n uri: " url "\n opts: " opts)))))
+
+
+(defn json->edn [s]
+  (json/parse-string s true))
+
+(defn invoke
```

Comment:
Sunt funcții aici care nu țin de CLI. 
Ar trebui separate în ns-ul lor. 

Ai API clojure.
Ai api CLI care folosește API clojure să expună interfața clojure în CLI.
Mai putem avea API rest care face același lucru - ca serviciu REST.
etc.